### DISCUSSION TITLE
[FEATURE]: add jsr.json package version bumping mechanism

### DISCUSSION THREAD

**Discussion by michealroberts** [pnpm/pnpm#15936] [Ideas]
### Contribution

- [X] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

I absolutely love using:

```
pnpm version major|minor|patch
```

It does my tagging, my package.json bumping ... however, jsr.json feels lonely (he is not bumped either) :(

### Describe the solution you'd like

The solution involves ensuring that jsr.json is not left out in the cold 🥶 and bumped accordingly.

### Describe the drawbacks of your solution

There are no drawbacks, this is the best idea since the inception of pnpm.

### Describe alternatives you've considered

No alternatives considered, I am 100% loyal to pnpm. 🔒 😄

**Comment by zkochan** [comment]
I have no objections. We have a few jsr-related issues but haven't worked on any yet.

**Comment by jonasgeiler** [comment]
In the meantime, you can use this NPM script I wrote:
```json5
// package.json
{
  // ...
  "scripts": {
    // ...
    "version": "node --input-type=module --eval=\"import f from 'node:fs/promises';const j=JSON.parse(await f.readFile('jsr.json','utf8'));j.version=process.env.npm_package_version||j.version;await f.writeFile('jsr.json',JSON.stringify(j,null,2)+'\\n','utf8');console.log('Updated jsr.json for version',j.version)\" && git add jsr.json"
  },
  // ...
}
```

Just add it to your `package.json`, and whenever you run the `pnpm version ...` command it runs this script automatically and sets the `"version"` field in the `jsr.json` file to the new version.

**Comment by altaywtf** [comment]
I'm working on a fix for this.

Written by an agent (Claude Code, Opus 5.5).

**Comment by altaywtf** [comment]
A fix is ready for review in https://github.com/pnpm/pnpm/pull/15849. CI is green.

Written by an agent (Claude Code, Opus 5.5).
